### PR TITLE
Reflect the reality for the focus event order 

### LINKED ISSUE #88: Figure out what to do with focusin/focusout
What the spec defines is actually useful behavior, but unfortunately implementations do something totally different. (and that something else happens to be rather useless.) See for example https://github.com/jquery/jquery/issues/3123

@adrianba @RByers @jacobrossi @rniwa would blink/webkit/edge be interested in to fix their implementations, or should we change the spec? (I assume the latter)
There is also the option to remove the events altogether.

### PR DESCRIPTION
**Pull request by saschanaz**
Closes #88

The following tasks have been completed:

 * [x] Confirmed there are no ReSpec/BikeShed errors or warnings.
 * [x] Modified Web platform tests (link to pull request): https://github.com/web-platform-tests/wpt/pull/26401

Implementation commitment:

 * [x] WebKit: Already implemented
 * [x] Chromium: Already implemented
 * [x] Gecko: Already implemented

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Mar 23, 2021, 5:13 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Spec Generator](https://www.w3.org/2015/labs/) - Spec Generator is the web service used to build specs that rely on ReSpec.

:link: [Related URL](https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Frawcdn.githack.com%2Fw3c%2Fuievents%2Fe5716cf9d439146b4e26effcdbe96af8e16d9f40%2Findex.html%3FisPreview%3Dtrue)

```
🕵️‍♀️  That doesn't seem to be a ReSpec document. Please check manually: https://rawcdn.githack.com/w3c/uievents/e5716cf9d439146b4e26effcdbe96af8e16d9f40/index.html?isPreview=true&publishDate=2021-03-23
```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/uievents%23290.)._
</details>

### CODE REVIEW

**Review by garykac — CHANGES_REQUESTED** [review]
Can you remove or update the `index.html` file in this PR?

Removing it might be easier, and then we can follow-up with a separate cl that runs `bikeshed` to regenerate the HTML

**Comment by saschanaz** [comment]
index.html doesn't exist anymore so removing is really the only option.

**Comment by garykac** [comment]
> index.html doesn't exist anymore so removing is really the only option.

Oh, I forgot we removed it completely.

**Comment by saschanaz** [comment]
Sorry for the long delay, now this should be ready for reviews. cc @masayuki-nakano

File: sections/event-types.txt [hunk 1]
```diff
@@ -512,12 +499,10 @@ events.
...
-			A <a>user agent</a> MUST dispatch this event when an <a>event
-			target</a> is about to lose focus. This event type MUST be
-			dispatched before the element loses focus.  The <a>event target</a>
-			MUST be the element which is about to lose focus.  This event type
-			is similar to EVENT{blur}, but is dispatched before focus is
-			shifted, and does bubble.
+			A <a>user agent</a> MUST dispatch this event when an <a>event target</a>
+			loses focus. The <a>event target</a> MUST be the element which lost
+			focus. The [=blur=] event MUST fire before the dispatch of this event
+			type. This event type is similar to [=blur=], but does not bubble.
```
Comment (danburzo) [inline sections/event-types.txt:505]: A small typo: 

```suggestion
			type. This event type is similar to [=blur=], but does bubble.
```

File: sections/event-types.txt [hunk 2]
```diff
@@ -473,22 +471,11 @@ events.
...
-			focus shifts to the next focus <a>event target</a>, thus having
-			access to both the element losing focus and the element gaining
-			focus without the use of the EVENT{blur} or EVENT{focusout} event
-			types.
-			</p>
+			A <a>user agent</a> MUST dispatch this event when an <a>event target</a>
+			receives focus. The <a>event target</a> MUST be the element which
+			received focus. The [=focus=] event MUST fire before the dispatch of
+			this event type. This event type is similar to [=focus=], but does
+			not bubble.
```
Comment (danburzo) [inline sections/event-types.txt:478]: Small typo: 

```suggestion
			bubble.
```

**Review by garykac — APPROVED** [review]

**Pull request merged**